x-pack/legacy/plugins/ml/public/application/routing/routes/timeseriesexplorer.tsx

```diff
@@ -221,6 +231,9 @@ export const TimeSeriesExplorerUrlStateManager: FC<TimeSeriesExplorerUrlStateMan
       selectedJob !== undefined &&
       selectedForecastId !== undefined
     ) {
+      if (selectedForecastIdProp !== selectedForecastId) {
+        setSelectedForecastIdProp(undefined);
```

Comment:
this state update will cause the re-render, then you call `getForecastDateRange` with `selectedForecastId ` and update the state again. Is it necessary? 
And why don't you update the appState instead?

Reply:
It's a bit fiddly but it's the way I was able to fix it without a major rewrite of the inner workings of `timeseriesexplorer.js` :). The combination of `selectedForecastId` (source of truth in the URL) and `selectedForecastIdProp` allows me track if a forecast update was completed. Otherwise multiple independent `useEffect`s would just trigger a bunch of rerenders and we cannot tell within `timeseriesexplorer.js` if we got everything we need for a forecast update.

The main pain point is that the inner charts still trigger a programmatic zoom update and with the current code there's no way to tell if that was triggered by a user or by another update. Fixing this would have required a much larger refactor.